lib/smart_proxy_openscap/openscap_lib.rb

```diff
@@ -13,10 +13,35 @@
 require 'json'
 require 'proxy/error'
 require 'proxy/request'
+require 'smart_proxy_openscap/openscap_exception'
 
 module Proxy::OpenSCAP
   extend ::Proxy::Log
 
+  def self.get_policy_content(policy_id)
+    policy_store_dir = File.join(Proxy::OpenSCAP::Plugin.settings.contentdir, policy_id.to_s)
+    policy_scap_file = File.join(policy_store_dir, "#{policy_id}_scap_content.xml")
+    FileUtils.mkdir_p(policy_store_dir) # will fail silently if exists
+
+    return File.open(policy_scap_file, 'rb').read if File.file?(policy_scap_file) && !File.zero?(policy_scap_file)
+
+    lock = Proxy::HttpDownloads.try_locking(policy_scap_file)
+
+    response = fetch_scap_content_xml(policy_id, policy_scap_file)
+    if lock.nil?
+      return response
+    else
+      begin
+        File.open(policy_scap_file, 'wb') do |file|
+          file << response
+        end
+      ensure
+        Proxy::HttpDownloads.unlock(lock)
+      end
+      return File.open(policy_scap_file, 'rb').read
```

Comment:
whats the difference to line 26 (https://github.com/OpenSCAP/smart_proxy_openscap/pull/10/files#diff-7b64de0f038ec2dbc5994deb74936428R26) ? if its the same extract to a method?